modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -484,22 +484,22 @@ private CompletableFuture<?> processOperationRequest(ReplicaRequest request, @Nu
      */
     private CompletableFuture<LeaderOrTxState> processTxStateCommitPartitionRequest(TxStateCommitPartitionRequest request) {
         return placementDriver.getPrimaryReplica(replicationGroupId, hybridClock.now())
-                .thenCompose(primaryReplica -> {
-                    if (primaryReplica == null) {
-                        return failedFuture(new PrimaryReplicaMissException(localNode.name(), null));
+                .thenCompose(primaryReplicaMeta -> {
+                    if (primaryReplicaMeta == null) {
+                        return failedFuture(new PrimaryReplicaMissException(localNode.name(), null, null, null, null));
                     }
 
-                    if (isLocalPeer(primaryReplica.getLeaseholder())) {
-                        TransactionMeta txMeta = txManager.stateMeta(request.txId());
+                    if (!isLocalPeer(primaryReplicaMeta.getLeaseholder())) {
```

Comment:
Why?

Reply:
For me, it reads better:
1. if the meta is zero, then there is an error
2. if the replica is not a local node, then an immediate response
3. something more complicated

I also find it easier to read because you keep less context in head.

Reply:
fair enough